test/e2e/pivoting_test.go

```diff
@@ -258,32 +325,62 @@ func labelHDCRDs(targetCluster framework.ClusterProxy) {
 	for _, label := range labels {
 		var cmd *exec.Cmd
 		if kubectlArgs == "" {
-			cmd = exec.Command("kubectl", "label", "--overwrite", "crds", crdName, label)
+			cmd = exec.Command("kubectl", "label", "--overwrite", "crds", crdName, label) // #nosec G204:gosec
 		} else {
-			cmd = exec.Command("kubectl", kubectlArgs, "label", "--overwrite", "crds", crdName, label)
+			cmd = exec.Command("kubectl", kubectlArgs, "label", "--overwrite", "crds", crdName, label) // #nosec G204:gosec
 		}
 		err := cmd.Run()
 		Expect(err).To(BeNil(), "Cannot label HD CRDs")
 	}
 }
 
-func rePivoting() {
+type RePivotingInput struct {
+	E2EConfig             *clusterctl.E2EConfig
+	BootstrapClusterProxy framework.ClusterProxy
+	TargetCluster         framework.ClusterProxy
+	SpecName              string
+	ClusterName           string
+	Namespace             string
+	ArtifactFolder        string
+	ClusterctlConfigPath  string
+}
+
+func rePivoting(ctx context.Context, inputGetter func() RePivotingInput) {
 	Logf("Start the re-pivoting test")
+	input := inputGetter()
 	By("Remove Ironic deployment from target cluster")
-	removeIronicDeploymentOnTarget(targetCluster)
-
+	removeIronic(ctx, func() RemoveIronicInput {
+		return RemoveIronicInput{
+			ManagementCluster: input.TargetCluster,
+			IsDeployment:      true,
+			Namespace:         input.E2EConfig.GetVariable(ironicNamespace),
+			NamePrefix:        input.E2EConfig.GetVariable(NamePrefix),
+		}
+	})
 	By("Reinstate Ironic containers and BMH")
+	// TODO: add this case to installIronicBMO function
```

Comment:
I am not sure if this is needed at all, if so, can you move it to the spefic func and make it clear what do we mean by "this case"

Reply:
here in this case we deploy ironic locally using `run_local_ironic.sh` in the function `installIronicBMO` we deploy ironic or bmo or both as deployment using `deploy.sh` I think the same function can cover all the case I do not want to have multiple function deploying ironic that becomes confusing I will do this changes in a separate PR since it is not tight to the targeted changes here (using function inputs and removing file suffix)